stratum/lib/security/credentials_manager.cc

```diff
@@ -13,26 +13,136 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-
 #include "stratum/lib/security/credentials_manager.h"
 
+#include <memory>
+#include <string>
+
+#include "absl/memory/memory.h"
 #include "gflags/gflags.h"
+#include "grpcpp/security/server_credentials.h"
+#include "grpcpp/security/server_credentials_impl.h"
+#include "grpcpp/security/tls_credentials_options.h"
 #include "stratum/glue/logging.h"
-#include "absl/memory/memory.h"
+#include "stratum/glue/status/status.h"
+#include "stratum/lib/macros.h"
+#include "stratum/lib/utils.h"
 
-namespace stratum {
+DEFINE_string(ca_cert, "", "CA certificate path");
+DEFINE_string(server_key, "", "gRPC Server pricate key path");
+DEFINE_string(server_cert, "", "gRPC Server certificate path");
 
-CredentialsManager::CredentialsManager() {}
+namespace stratum {
 
 CredentialsManager::~CredentialsManager() {}
+CredentialsManager::CredentialsManager() {}
 
 std::shared_ptr<::grpc::ServerCredentials>
 CredentialsManager::GenerateExternalFacingServerCredentials() const {
-  return ::grpc::InsecureServerCredentials();
+  return server_credentials_;
 }
 
 std::unique_ptr<CredentialsManager> CredentialsManager::CreateInstance() {
-  return absl::WrapUnique(new CredentialsManager());
+  auto instance_ = absl::WrapUnique(new CredentialsManager());
+  auto status = instance_->Initialize();
+  if (!status.ok()) {
+    LOG(ERROR) << "Failed to initialize the CredentialsManager instance: "
+               << status.error_message();
+    return nullptr;
```

Comment:
I have looked at the call sites for this and none check against `nullptr`.

I'd suggest
- Make `Initialize` private and call with `RETURN_IF_ERROR`
- Change the return type of `CreateInstance` to ::utils::StatusOr<std::unique_ptr<CredentialsManager>>
- Use ASSIGN_OR_RETURN in all call sites